numpy/_core/src/umath/ufunc_object.c

```diff
@@ -1590,6 +1590,13 @@ _get_coredim_sizes(PyUFuncObject *ufunc, PyArrayObject **op,
         }
     }
 
+    if (ufunc->process_core_dims_func != NULL) {
+        int status = ufunc->process_core_dims_func(ufunc, core_dim_sizes);
+        if (status != 0) {
+            return -1;
+        }
+    }
+
```

Comment:
So one thing we do *not* do here is sanity check the output.  I.e. if the `core_dim_size` is already set but the user mutates it, we could be in trouble.

To be clear, I think that is OK, but it should be documented.

Reply:
It is easy enough to pass a copy to `process_core_dims_func` and verify on return that nothing was changed that shouldn't be changed.  I'll add that.

Reply:
I don't think it matters, just occured to me that we should document that you may only validate (raise an error) or fill in `-1`.

Reply:
I wavered a bit about whether or not to test, but in the end, making a mistake in the processing here is no different from making a mistake in the actual function that gets called - one can get a core dump regardless. So, agreed with @seberg that a check after the call is not needed.

Reply:
OK, I *won't* add the check!